pkg/cloudprovider/cloudprovider.go

```diff
@@ -165,6 +169,50 @@ func (c *CloudProvider) Delete(ctx context.Context, node *v1.Node) error {
 	return c.instanceProvider.Terminate(ctx, node)
 }
 
+func (c *CloudProvider) IsMachineDrifted(ctx context.Context, machine *corev1alpha1.Machine) (bool, error) {
```

Comment:
This logic is going to grow as we add in things like instanceProfile, securityGroup, etc. How do you see this evolving?

Reply:
Putting notes from our discussion here: 
- Moving most of the AMI logic to a separate function. 
- Maybe we can move some of the additional drift functions as a sort of requirements so that we can easily cache resolved values of the AWSNodeTemplate settings as some sort of easily comparable values.